sample-tracking-status-overview-domain/src/main/groovy/life/qbic/business/samples/download/DownloadManifest.groovy

```diff
@@ -0,0 +1,46 @@
+package life.qbic.business.samples.download
+/**
+ * <b>A download manifest for download using our postman-cli</b>
+ *
+ * <p>Represents a download manifest. This manifest can be printed and used with the <a href="https://github.com/qbicsoftware/postman-cli#provide-a-file-with-several-qbic-ids">postman-cli.</a></p>
+ *
+ * @since 1.0.0
+ */
+class DownloadManifest {
+
+    private Set<String> sampleCodes
+
+    /**
+     * Please use {@link DownloadManifest#from(Collection<String>)} to create a DownloadManifest
+     * @see DownloadManifest#from
+     */
+    private DownloadManifest() {
+        sampleCodes = new LinkedHashSet<>()
+    }
+
+    private static String printSampleRow(String sampleCode) {
+        return "${sampleCode}\n"
+    }
+
+
+    /**
+     * Creates a download manifest from the provided sample codes
+     * @param sampleCodes a list of sample codes that are contained in this manifest
+     * @return a DownloadManifest containing the provided sample codes
+     * @since 1.0.0
+     */
+    static DownloadManifest from(Collection<String> sampleCodes) {
+        DownloadManifest downloadManifest = new DownloadManifest()
+        downloadManifest.sampleCodes.addAll(sampleCodes.unique())
+        return downloadManifest
+    }
+
+    /**
+     * Print the download Manifest
+     * @return a multi-line String print of the download manifest
+     * @since 1.0.0
+     */
+    String print() {
+        sampleCodes.collect {printSampleRow(it)}.sum()
+    }
```

Comment:
This is rather a toString then, isn't it?

Reply:
No `toString` states that it is not meant in those cases explicitly. This is an anti-pattern

Reply:
I understand, it does not necessarily need to be called like that. I just wanted to emphasise why it is confusing to me

Reply:
But your point is valid. It is the string representation at this moment.